core/src/org/labkey/core/admin/AdminController.java

```diff
@@ -10946,18 +10945,30 @@ public class ContentSecurityPolicyReportAction extends ReadOnlyApiAction<SimpleA
         @Override
         public Object execute(SimpleApiJsonForm form, BindException errors) throws Exception
         {
+            var ret = new JSONObject().put("success",true);
+
+            // fail fast
+            if (!_log.isWarnEnabled())
+                return ret;
+
             // NOTE User will always be "guest".  Seems like a bad design to force the server to accept guest w/o CSRF here.
             var jsonObj = form.getJsonObject();
             if (null != jsonObj)
             {
                 var jsonStr = jsonObj.toString();
-                JSONObject cspReport = jsonObj.getJSONObject("csp-report");
-                String urlString = cspReport.getString("source-file");
-                String path = new URLHelper(urlString).deleteParameters().getPath();
-                if (null == reports.put(path,Boolean.TRUE))
-                    _log.warn("ContentSecurityPolicy warning:\n" + jsonStr);
+                JSONObject cspReport = jsonObj.optJSONObject("csp-report");
+                if (cspReport != null)
+                {
+                    String urlString = cspReport.optString("document-uri");

```

Comment:
Unless specified, `optString(key)`'s default value is `""`, not `null`.
```suggestion
                    String urlString = cspReport.optString("document-uri", null);
```

`org.json` has numerous questionable design decisions with regard to `null` values.

Reply:
Thanks, now I know.